src/xunit.console.uwp/CommandLine.cs

```diff
@@ -0,0 +1,149 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Xunit.Abstractions;
+
+namespace Xunit.UwpClient
+{
+    internal class CommandLine
+    {
+        readonly Stack<string> arguments = new Stack<string>();
+
+        protected CommandLine(string[] args, Predicate<string> fileExists = null)
+        {
+            if (fileExists == null)
+                fileExists = File.Exists;
+
+            for (var i = args.Length - 1; i >= 0; i--)
+                arguments.Push(args[i]);
+
+            Project = Parse(fileExists);
+        }
+
+        public bool Debug { get; protected set; }
+
+        public bool NoLogo { get; protected set; }
+
+        public XunitProject Project { get; protected set; }
+
+        public bool Wait { get; protected set; }
+        public string InstallLocation { get; internal set; }
+
+        protected virtual string GetFullPath(string fileName)
+        {
+            return Path.GetFullPath(fileName);
+        }
+
+        XunitProject GetProjectFile(List<Tuple<string, string>> assemblies)
+        {
+            var result = new XunitProject();
+
+            foreach (var assembly in assemblies)
+                result.Add(new XunitProjectAssembly
+                {
+                    AssemblyFilename = GetFullPath(assembly.Item1),
+                    ConfigFilename = assembly.Item2 != null ? GetFullPath(assembly.Item2) : null,
+                });
+
+            return result;
+        }
+
+        static void GuardNoOptionValue(KeyValuePair<string, string> option)
+        {
+            if (option.Value != null)
+                throw new ArgumentException($"error: unknown command line option: {option.Value}");
+        }
+
+        static bool IsConfigFile(string fileName)
+        {
+            return fileName.EndsWith(".config", StringComparison.OrdinalIgnoreCase)
+                || fileName.EndsWith(".json", StringComparison.OrdinalIgnoreCase);
+        }
+
+        public static CommandLine Parse(params string[] args)
```

Comment:
Most of this code is duplicated between here and the console app.  Could we unify this code and share it between the two?